crates/miden-objects/src/block/nullifier_tree.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+    block::BlockNumber,
+    crypto::{
+        hash::rpo::RpoDigest,
+        merkle::{MutationSet, Smt, SmtProof, SMT_DEPTH},
+    },
+    errors::NullifierTreeError,
+    note::Nullifier,
+    Felt, FieldElement, Word,
+};
+
+/// Nullifier SMT.
+#[derive(Debug, Clone)]
+pub struct NullifierTree(Smt);
```

Comment:
What do we use this struct for?

Reply:
I think overall it makes sense to have `PartialNullifierTree` and `NullifierTree` (as well as analogous wrapper types for the account tree later) in miden-base. That way, we can use them in the MockChain and don't have to deal with raw `Smt`s instead and re-implement protocol logic (e.g. nullifier value layout) in multiple places. Would you agree?

For now we don't use either of them in this PR, because the root updates are computed in the `LocalBlockProver` which I'll add in a separate PR. So I'll move them there instead.